### PR TITLE
feat(types): Support for oauth_atlassian

### PR DESCRIPTION
**Pull request by yourtallness**
## Type of change

- [ ] 🐛 Bug fix
- [X] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [X] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [X] `npm test` runs as expected.
- [X] `npm run build` runs as expected.

## Related issue

https://www.notion.so/clerkdev/Sign-in-with-Atlassian-b3dcb47718774e40ba7cacfa47d8d17e

### CODE REVIEW

File: packages/types/src/oauth.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export type TwitterOauthProvider = 'twitter';
 export type TwitchOauthProvider = 'twitch';
 export type LinkedinOauthProvider = 'linkedin';
 export type DropboxOauthProvider = 'dropbox';
+export type AtlassianBitbucketProvider = 'atlassian';
```
Comment (chanioxaris) [inline packages/types/src/oauth.ts:21]:
```suggestion
export type AtlassianOAuthProvider = 'atlassian';
```

File: packages/types/src/oauth.ts [hunk 2]
```diff
@@ -39,6 +40,7 @@ export type OAuthProvider =
   | TwitchOauthProvider
   | LinkedinOauthProvider
   | DropboxOauthProvider
+  | AtlassianBitbucketProvider
```
Comment (chanioxaris) [inline packages/types/src/oauth.ts:43]:
```suggestion
  | AtlassianOAuthProvider
```

File: packages/types/src/oauth.ts (hunk 1 above)
Comment (yourtallness) [reply packages/types/src/oauth.ts]: Thanks, fixed

File: packages/types/src/oauth.ts (hunk 2 above)
Comment (yourtallness) [reply packages/types/src/oauth.ts]: Thanks, fixed

**Pull request merged**